department-of-veterans-affairs/vets-website
# 12057 adds hook for staggered feature release

## Summary

Allows for staggered release of soft launch modal via reusable hook. Specifically, this PR utilizes the hook to allow a percentage of visitors to be shown the soft-launch modal for the new hompage, currenrtly set at 25%.
As similar functionality was used in the past for the chat-bot, we've opted to implement this functionality as a custom React hook for future reuse. 

## Related issue(s)
- department-of-veterans-affairs/va.gov-cms/issues/12057



## Testing done

- Manual Testing completed.

## Screenshots
n/a - no visual aspect.


## What areas of the site does it impact?
Currently, this will effect 25% of users visiting the homepage, once the feature flag is enabled.

## Acceptance criteria

- [ ]  Events are being sent to the appropriate logging solution - Handled by department-of-veterans-affairs/va.gov-cms/issues/12048
- [x]  Documentation has been updated (see doc block above new React hook)
- [n/a]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [n/a]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [n/a]  I added a screenshot of the developed feature

## QA Steps
1. Build `vets website` with the following command: `yarn watch --env api=https://staging-api.va.gov` or visit the test environment.
2. There's a 25% chance you'll be shown the soft-launch modal automatically. If you aren't, check the localStorage for the key of `show-homepage-soft-launch-modal` and set the value to be below or equal to 25. 
3. Refresh the page. You should now see the modal. 
4.  Keep in mind, if you dismiss the modal at any point, this dismissal will be persisted in sessionStorage. Clear sessionStorage or repeat steps in a new tab to verify functionality more than once.
5. Try changing the localStorage value to a number greater than 25 and see that you're *not* shown the modal.

## Requested Feedback

Assure modal functions as desired according to the QA Steps above.
